packages/eslint-plugin/docs/rules/no-explicit-any.md

```diff
@@ -158,7 +158,24 @@ interface Garply {
 
 ## When Not To Use It
 
-If an unknown type or a library without typings is used
+**Higher Order Functions**. If you're typing a higher order function such as `compose`, `pipe`, etc, you may need `any` because TypeScript types are not fully co-expressive with idiomatic functional JavaScript. Such functions are intentionally generic, and you can assign explicit types later to the returned functions, e.g.:
```

Comment:
> fully co-expressive with idiomatic functional JavaScript

The vast majority of our readers won't know what this means 🙂. Let's stick with phrases that keep docs approachable for newcomers.

IMO docs generally don't need to be exactly precise for topics that don't need it. I think it'd be ok here to lose a little precision and go with something like:

```diff
- TypeScript types are not fully co-expressive with idiomatic functional JavaScript.
+ TypeScript types can't always fully represent functional JavaScript patterns.
```

Thoughts?

(I'll separately comment that I think we can make this more generalized - this is more a discussion for the fun of it than anything else)

Reply:
Huge improvement!